sw/device/tests/aon_timer_wakeup_irq_test.c

```diff
@@ -0,0 +1,177 @@
+  uint32_t sleep_range_l = (float)sleep_time_us * 0.85;
+
+  // Setup the timer and wait for the IRQ
+  uint32_t sleep_cycles = (sleep_time_us * kClockFreqAonHz / 1000000);
+  LOG_INFO("Setting wakeup interrupt for %u us (%u cycles)", sleep_time_us,
+           sleep_cycles);
+  // Capture the current tick to measure the time the IRQ will take
+  time_elapsed = tick_count_get();
+  aon_timer_testutils_wakeup_config(aon_timer, sleep_cycles);
+  do {
+    wait_for_interrupt();
+  } while (peripheral != kTopEarlgreyPlicPeripheralAonTimerAon &&
+           time_elapsed < sleep_range_h);
+
+  CHECK(time_elapsed < sleep_range_h && time_elapsed > sleep_range_l,
+        "Timer took %u usec which is not in the range %u usec and %u usec",
+        (uint32_t)time_elapsed, sleep_range_l, sleep_range_h);
+
+  CHECK(peripheral == kTopEarlgreyPlicPeripheralAonTimerAon,
+        "Interrupt from incorrect peripheral: exp = %d, obs = %d",
+        kTopEarlgreyPlicPeripheralAonTimerAon, peripheral);
+
+  CHECK(irq == kDifAonTimerIrqWkupTimerExpired,
+        "Interrupt type incorrect: exp = %d, obs = %d",
+        kDifAonTimerIrqWkupTimerExpired, irq);
+}
+
+/**
+ * External interrupt handler
+ */
+void ottf_external_isr(void) {
+  // Calc the elapsed time since the activation of the IRQ
+  time_elapsed = tick_count_get() - time_elapsed;
+
+  dif_rv_plic_irq_id_t irq_id;
+  CHECK_DIF_OK(dif_rv_plic_irq_claim(&plic, kPlicTarget, &irq_id));
+
+  irq = (dif_aon_timer_irq_t)(
+      irq_id -
+      (dif_rv_plic_irq_id_t)kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired);
+
+  peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[irq_id];
+
+  CHECK_DIF_OK(dif_aon_timer_irq_acknowledge(&aon_timer, irq));
+  // Complete the IRQ by writing the IRQ source to the Ibex specific CC
+  // register.
+  CHECK_DIF_OK(dif_rv_plic_irq_complete(&plic, kPlicTarget, irq_id));
+}
+
+bool test_main(void) {
+  // Enable global and external IRQ at Ibex.
+  irq_global_ctrl(true);
+  irq_external_ctrl(true);
+
+  // Initialize the rv timer to compute the tick
+  tick_init();
+
+  // Initialize pwrmgr
+  dif_pwrmgr_t pwrmgr;
+  CHECK_DIF_OK(dif_pwrmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR), &pwrmgr));
+
+  // Initialize aon timer
+  CHECK_DIF_OK(dif_aon_timer_init(
+      mmio_region_from_addr(TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR), &aon_timer));
+
+  dif_pwrmgr_irq_state_snapshot_t pwrmgr_irqs =
+      (dif_pwrmgr_irq_state_snapshot_t)UINT_MAX;
+
+  CHECK_DIF_OK(dif_pwrmgr_irq_restore_all(&pwrmgr, &pwrmgr_irqs));
+
+  // Initialize the PLIC.
+  mmio_region_t plic_base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR);
+  CHECK_DIF_OK(dif_rv_plic_init(plic_base_addr, &plic));
+
+  // Enable all the AON interrupts
+  rv_plic_testutils_irq_range_enable(
+      &plic, kPlicTarget, kTopEarlgreyPlicIrqIdPwrmgrAonWakeup,
+      kTopEarlgreyPlicIrqIdAonTimerAonWdogTimerBark);
```

Comment:
Doesn't seem right, as you surely don't need all of the following interrupts enabled for this test?

```
  kTopEarlgreyPlicIrqIdPwrmgrAonWakeup = 154, /**< pwrmgr_aon_wakeup */
  kTopEarlgreyPlicIrqIdSysrstCtrlAonSysrstCtrl = 155, /**< sysrst_ctrl_aon_sysrst_ctrl */
  kTopEarlgreyPlicIrqIdAdcCtrlAonDebugCable = 156, /**< adc_ctrl_aon_debug_cable */
  kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired = 157, /**< aon_timer_aon_wkup_timer_expired */
  kTopEarlgreyPlicIrqIdAonTimerAonWdogTimerBark = 158, /**< aon_timer_aon_wdog_timer_bark */
```

Reply:
I don't see an issue here since I'm checking in the test whether the correct interrupt fired.

Reply:
Well, there is no point enabling interrupts that you don't need for the test, as it is confusing.

Reply:
It doesn't hurt to configure and enable ALL interrupts from an IP even if only one interrupt is expected to fire. Indeed, what is verified by doing so is the right interrupt from AON timer is wired in hardware correctly to the right interrupt index at PLIC. Perhaps worth clarifying this in a comment.

Reply:
Makes sense, I agree that the selection of interrupts should be documented, otherwise a reader might think it was a mistake.

Reply:
I changed to enable just the interruptions related to the AON Timer (157 and 158).